scrapy_poet/injection.py

```diff
@@ -169,10 +178,14 @@ def build_instances(self, request: Request, response: Response, plan: andi.Plan)
 
     @inlineCallbacks
     def build_instances_from_providers(
-        self, request: Request, response: Response, plan: andi.Plan
+        self,
+        request: Request,
+        response: Response,
+        plan: andi.Plan,
+        prev_instances: Optional[Dict] = None,
     ):
         """Build dependencies handled by registered providers"""
-        instances: Dict[Callable, Any] = {}
+        instances: Dict[Callable, Any] = prev_instances or {}
```

Comment:
Passing the prev instances allows the PO to gain access to the item produced by a provider.

```python
@attrs.define 
class ProductPage(WebPage[Product]):
    product: Product
```

In the above example, before the `ProductPage` instance can be made, the `product: Product` instance should first be created. This is where this passing of new `prev_instances` parameter is useful.